delphi/translators/for2py/genPGM.py

```diff
@@ -3474,18 +3498,52 @@ def create_grfn_dict(
     generator.fortran_file = original_file
 
     try:
-        filename_regex = re.compile(r"(.*)\..*$")
-        name_match = re.match(filename_regex, file_name)
-        assert name_match, f"Can't match filename to any format: {file_name}"
-        file_name = name_match.group(1)
+        filename_regex = re.compile(r"(?P<path>.*/)(?P<filename>.*).py")
+        file_match = re.match(filename_regex, file_name)
+        assert file_match, f"Can't match filename to any format: {file_name}"
+
+        path = file_match.group("path")
+        filename = file_match.group("filename")
+
+        # Since we do not have separate variable pickle file
+        # for m_*.py, we need to use the original program pickle
+        # file that module resides.
+        module_name = None
+        if module_file_exist:
+            module_file_path = file_name
+            module_name = filename[2:]  # First two elements are "m_"
```

Comment:
The assumption about "m_" prefix is brittle.  I would like to minimize our reliance on it.